### ISSUE TITLE
`--help` flag missing on `extern` commands

### ISSUE THREAD

**Issue by theAkito** [nushell/nushell#7207]
### Describe the bug

```nu
$ zpool import --help
Error: nu::parser::unknown_flag (link)

  × The `zpool import` command doesn't have flag `help`.
   ╭─[entry #13:1:1]
 1 │ zpool import --help
   ·              ───┬──
   ·                 ╰── unknown flag
   ╰────
  help: use zpool import --help for a list of flags
```

### How to reproduce

1. Create `extern "zpool import" [............]`.
2. Run `zpool import` with `--help`, as suggested by Nu.
3. Ironically, see Nu telling you to run `zpool import --help` when exactly that does not work.

### Expected behavior

It shouldn't tell the user to use the `--help` flag when it clearly isn't available. Especially, when the command is failing due to having just used the non-existent `--help` flag.

### Configuration

| key                | value                                           |
| ------------------ | ----------------------------------------------- |
| version            | 0.71.0                                          |
| branch             |                                                 |
| commit_hash        | 6cc4ef6c706ebbdc5b995013e0b00f77f7a4dba7        |
| build_os           | linux-x86_64                                    |
| build_target       | x86_64-unknown-linux-gnu                        |
| rust_version       | rustc 1.65.0 (897e37553 2022-11-02)             |
| rust_channel       | stable-x86_64-unknown-linux-gnu                 |
| cargo_version      | cargo 1.65.0 (4bc8f24d3 2022-10-20)             |
| pkg_version        | 0.71.0                                          |
| build_time         | 2022-11-08 18:00:26 +00:00                      |
| build_rust_channel | release                                         |
| features           | database, dataframe, default, trash, which, zip |
| installed_plugins  | partial                                         |

### Additional context

Similar to issues like https://github.com/nushell/nushell/issues/6020, but not the same. This one is focused on the irony of Nu suggesting the use of `--help`, when the command just failed executing with that flag.

Either a `--help` flag should actually be generated, or the suggestion of using `--help` should be improved or removed for the sake of keeping the suggestions consistent with what is helpful and actually makes sense.

**Comment by sholderbach** [comment]
Yeah, there are many issues in our backlog that describe problems resulting from us currently using the `extern` definition to do preemptive argument validation. 
I think this should be a warning sign that we should let the external tool handle the unknown flag case themselves.
(I shudder at the amount of issues to go through to find those to close if we decide to change that)

**Comment by 132ikl** [comment]
This is no longer an issue on 0.107.0
